### PR TITLE
modal bottom sheet barrier label

### LINKED ISSUE #83180: [Semantic] showModalBottomSheet with custom barrierLabel
## Proposal

Allow `showModalBottomSheet` to accept custom `barrierLabel` to make the modalSheet more descriptive for the users with ScreenReaders.

### PR DESCRIPTION
**Pull request by NikolajHarderNota**
Adds barrierLabel as optional param in showModalBottomSheet

Fixes #83180

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by QuncCccccc** [comment]
Hi @NikolajHarderNota ! Thanks a lot for the contribution! There seems some failed tests, could you help fix them?
Screenshot 2023-05-12 at 3 09 10 PM

File: packages/flutter/test/material/bottom_sheet_test.dart [hunk 1]
```diff
@@ -1999,6 +1999,97 @@ void main() {
...
+      showModalBottomSheet<void>(
+        barrierLabel: customLabel,
+        context: scaffoldKey.currentContext!,
+        routeSettings: routeSettings,
+        builder: (BuildContext context) {
+          retrievedRouteSettings = ModalRoute.of(context)!.settings;
+          return const Text('BottomSheet');
+        },
+      );
+debugPrint(retrievedRouteSettings.toString());
```
Comment (QuncCccccc) [inline packages/flutter/test/material/bottom_sheet_test.dart:2027]: Just curious, why do we need this here?

File: packages/flutter/test/material/bottom_sheet_test.dart [hunk 2]
```diff
@@ -1999,6 +1999,97 @@ void main() {
...
+      await tester.pump();
+      await tester.pump(const Duration(seconds: 1));
+
+
+      expect(retrievedRouteSettings, routeSettings);
```
Comment (QuncCccccc) [inline packages/flutter/test/material/bottom_sheet_test.dart:2032]: Should we check if the `customLabel` is correctly used here?

File: packages/flutter/test/material/bottom_sheet_test.dart [hunk 3]
```diff
@@ -1999,6 +1999,97 @@ void main() {
     });
 
   });
+
+    // ignore: always_declare_return_types
+  group('showModalBottomSheet modalBarrierDismissLabel', () {
```
Comment (QuncCccccc) [inline packages/flutter/test/material/bottom_sheet_test.dart:2004]: Looks like this group message is same as the below one, and the test is similar with the first test in the group below. Should these tests be put into one group, or maybe this group can just be removed?

File: packages/flutter/test/material/bottom_sheet_test.dart [hunk 4]
```diff
@@ -1999,6 +1999,97 @@ void main() {
     });
 
   });
+
+    // ignore: always_declare_return_types
```
Comment (QuncCccccc) [inline packages/flutter/test/material/bottom_sheet_test.dart:2003]: Should we remove this comment?Just wonder if this is needed.

File: packages/flutter/test/material/bottom_sheet_test.dart [hunk 5]
```diff
@@ -1999,6 +1999,97 @@ void main() {
...
+
+      expect(retrievedRouteSettings, routeSettings);
+      });
+  });
+
+  group('showModalBottomSheet modalBarrierDismissLabel', () {
+    testWidgets('Verify that barrierLabel is used if provided',
+        (WidgetTester tester) async {
+      final GlobalKey<ScaffoldState> scaffoldKey = GlobalKey<ScaffoldState>();
+      const RouteSettings routeSettings =
```
Comment (QuncCccccc) [inline packages/flutter/test/material/bottom_sheet_test.dart:2040]: Looks like this is not necessary for this test. Could you remove these(also the `routeSetting` in the test below) to make the tests simpler?

**Review by QuncCccccc — COMMENTED** [review]
Thanks a lot for helping add this feature! Overall it looks good. Just left some questions for the unit tests:) Also could you try to rebase the latest master to fix the "ci.yaml validation" check?

**Review by QuncCccccc — APPROVED** [review]
LGTM. Thanks for the contribution:)

**Review by reidbaker — APPROVED** [review]

**Pull request merged**